src/classes/UTIL_Permissions.cls

```diff
@@ -0,0 +1,294 @@
+    */
+    public static Boolean canCreate(String objectName) {
+        return canCreate(objectName, true);
+    }
+
+    /**
+    * @description Determines whether the running user has update permissions on the given object
+    * @param objectName the name of the object
+    * @return returns true if the user has permission; throws an exception if they don't
+    */
+    public static Boolean canUpdate(String objectName) {
+        return canUpdate(objectName, true);
+    }
+
+    /**
+    * @description Determines whether the running user has delete permissions on the given object
+    * @param objectName the name of the object
+    * @return returns true if the user has permission; throws an exception if they don't
+    */
+    public static Boolean canDelete(String objectName) {
+        return canDelete(objectName, true);
+    }
+
+    /**
+    * @description Determines whether the running user has read permissions on the given object
+    * @param objectName the name of the object
+    * @param throwException indicates whether an exception should be thrown if the user doesn't have permission
+    * @return returns boolean to indicate whether the user has permission; conditionally throws an exception if they don't
+    */
+    public static Boolean canRead(String objectName, Boolean throwException) {
+        if (!UTIL_Describe.getObjectDescribe(objectName).isAccessible()) {
+            if (throwException) {
+                throw new InsufficientPermissionException(INSUFFICIENT_PERMISSION_MESSAGE);
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+    * @description Determines whether the running user has create permissions on the given object
+    * @param objectName the name of the object
+    * @param throwException indicates whether an exception should be thrown if the user doesn't have permission
+    * @return returns boolean to indicate whether the user has permission; conditionally throws an exception if they don't
+    */
+    public static Boolean canCreate(String objectName, Boolean throwException) {
+        if (!UTIL_Describe.getObjectDescribe(objectName).isCreateable()) {
+            if (throwException) {
+                throw new InsufficientPermissionException(INSUFFICIENT_PERMISSION_MESSAGE);
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+    * @description Determines whether the running user has update permissions on the given object
+    * @param objectName the name of the object
+    * @param throwException indicates whether an exception should be thrown if the user doesn't have permission
+    * @return returns boolean to indicate whether the user has permission; conditionally throws an exception if they don't
+    */
+    public static Boolean canUpdate(String objectName, Boolean throwException) {
```

Comment:
I maybe have a slight preference for grouping the overloaded of canUpdate(), canCreate() etc to be next to each other in the code just for ease of reading and drawing that association, but not by any means a big deal